utilities/dataflow_kafka_to_hcapi/dataflow_kafka_to_hcapi.py

```diff
@@ -0,0 +1,529 @@
+#    Copyright 2023 Google LLC
+
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+
+#        http://www.apache.org/licenses/LICENSE-2.0
+
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+##################################
+# Author: Devansh Modi           #
+##################################
+
+#Main Dataflow MODULE :- fetching from PubSub, cleaning, filtering, transforming and posting to HCAPI
+
+#Importing Necessary Libraries
+from google.cloud import secretmanager
+import apache_beam as beam
+from apache_beam import pvalue
+from apache_beam import io
+from apache_beam.io.kafka import ReadFromKafka
+from apache_beam.options.pipeline_options import PipelineOptions
+import json
+import logging
+import argparse
+from datetime import datetime, timedelta
+import google.auth
+import google.auth.transport.requests
+import requests
+import base64
+import uuid
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+import configparser
+import time
+
+   
+#Below class is used to POST and GET FHIR resources from HealthCare API (We will use Patient FHIR resource as an example)
+class hcapi_fhir_store:
+    def __init__(self, args:dict):
+        self.hcapi_project_id = str(args['hcapi_project_id'])
+        self.hcapi_version = str(args['hcapi_version'])
+        self.hcapi_location = str(args['hcapi_location'])
+        self.hcapi_dataset = str(args['hcapi_dataset'])
+        self.hcapi_get_fhir_store = str(args['hcapi_get_fhir_store'])
+        self.hcapi_post_fhir_store = str(args['hcapi_post_fhir_store'])
+      
+
+    def google_api_headers(self):
```

Comment:
is this equivalent to running `gcloud auth application-default login`? I think it is but just want to confirm, otherwise, we can just tell the user to run the above `gcloud` command?

Reply:
Yes it is equivalent of that but the token expires in 1 hour. Thats where we are generating new token per API call based on frequency to authenticate and GET/POST FHIR resources.